huggingface_pipelines/pipeline_config.py

```diff
@@ -0,0 +1,179 @@
+        return replace(self, shard_id=shard_id)
+
+    def with_cache_to_arrow(self, cache_to_arrow: bool):
+        return replace(self, cache_to_arrow=cache_to_arrow)
+
+    def with_output_file_name(self, output_file_name: str):
+        return replace(self, output_file_name=output_file_name)
+
+    def with_columns(self, columns: List[str]):
+        return replace(self, columns=columns)
+
+    def with_take(self, take: int):
+        return replace(self, take=take)
+
+
+@dataclass
+class TextToEmbeddingPipelineConfig(PipelineConfig):
+    """
+    Configuration class for text-to-embedding pipelines.
+
+    Attributes:
+        encoder_model (str): The name or path of the model to be used for encoding texts into embeddings.
+        source_lang (str): The source language code for the texts to be encoded.
+    """
+    encoder_model: str = "text_sonar_basic_encoder"
+    source_lang: str = "eng_Latn"
+
+    def with_encoder_model(self, encoder_model: str):
+        return replace(self, encoder_model=encoder_model)
+
+    def with_source_lang(self, source_lang: str):
+        return replace(self, source_lang=source_lang)
+
+
+@dataclass
+class EmbeddingToTextPipelineConfig(PipelineConfig):
+    """
+    Configuration class for embedding-to-text pipelines.
+
+    Attributes:
+        decoder_model (str): The name or path of the model to be used for decoding embeddings back into texts.
+        target_lang (str): The target language code for the texts to be decoded.
+    """
+    decoder_model: str = "text_sonar_basic_decoder"
+    target_lang: str = "eng_Latn"
+
+    def with_decoder_model(self, decoder_model: str):
+        return replace(self, decoder_model=decoder_model)
+
+    def with_target_lang(self, target_lang: str):
+        return replace(self, target_lang=target_lang)
+
+
+@dataclass
+class AudioPipelineConfig(PipelineConfig):
+    """
+    Configuration class for ASR pipelines.
+
+    Attributes:
+        model_name (str): The name or path of the model to be used for transcribing audio to text.
+        audio_column (str): The column of the dataset containing audio data.
+        reference_transcriptions (str): The column of the dataset containing reference transcriptions.
+        data_file (str): The file containing audio data paths and metadata.
+        audio_root_dir (str): The root directory where audio files are stored.
+        audio_path_index (int): The index of the audio path in the data file.
+        target_lang (str): The target language for transcription.
+        pad_idx (int): Padding index for batching.
+        fbank_dtype (str): The data type for filter bank features.
+        n_parallel (int): Number of parallel workers.
+    """
+    encoder_model = "sonar_speech_encoder_eng"
+    decoder_model = "text_sonar_basic_decoder"
+    reference_transcriptions: str = "reference_transcription"
+    data_file: str = None
+    audio_root_dir: str = None
+    audio_path_index: int = None
+    target_lang: str = None
```

Comment:
should not they be mandatory ?
```suggestion
    target_lang: str
```